### PR TITLE
VaaS search, VaaS log, Teams VaaS/TPP

### LINKED ISSUE #73: Support Team Creation and Management
# Summary of the new feature/enhancement

Master-Admins and assigned users/groups are able to create and manage teams within TPP (GUI and API). VenafiPS should have the ability to get all teams, create/update/delete teams and manage team owners/members.

### PR DESCRIPTION
**Pull request by gdbarron**
- Add `New-VaasSearchQuery` as the framework for VaaS searching including filtering, ordering, and paging.  This will be used by certificate search, log search, and probably more in the future.
- Add `Read-VenafiLog` utilizing the new search framework.  Merge existing `Read-TppLog` into 1 function for VaaS and TPP.
- Add `Get-VenafiTeam` to retrieve all or specific team info, VaaS and TPP
- Add `New-VenafiTeam` to create a new team, VaaS and TPP
- Add `Remove-VenafiTeam` to remove a team, VaaS and TPP
- Add `Add-VenafiTeamMember` to add a team member, VaaS and TPP
- Add `Add-VenafiTeamOwner` to add a team owner, VaaS and TPP
- Add `Remove-VenafiTeamMember` to remove a team member, VaaS and TPP
- Add `Remove-VenafiTeamOwner` to remove a team owner, VaaS and TPP
- Add `ConvertTo-TppIdentity` to standardize TPP identity objects
- Add `Get-VenafiIdentity` to retrieve one identity, the current user, or all, VaaS and TPP.  This replaces `Get-TppIdentity`.
- Change `Invoke-TppRestMethod` to `Invoke-VenafiRestMethod` in remaining internal module calls
- Move key/token refresh messaging to TPP only in `VenafiSession` as no refresh for VaaS
- Default `-UriRoot` in `Invoke-VenafiRestMethod` to v1 for VaaS
closes #73

### CODE REVIEW

**Review by astro2000j — APPROVED** [review]
ignore...

**Comment by astro2000j** [comment]
New-VenafiTeam for TPP exposes the "Description" parameter. The function parameter expects an array, but the actual API call expects this to be a string.

Current:
```[string[]] $Description```

Expected:
```[string] $Description```

Code:
```New-VenafiTeam -Name "Team Alpha" -Product "TLS" -Owner @("AD+SECURAFI:1bec6c2b8dcdef4da8f6d78a80bd843f") -Member @("AD+SECURAFI:170c0a92645a17438d3c0718af825666") -Description "This is a test"```

File: VenafiPS/Public/New-VenafiTeam.ps1 [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        [Parameter(Mandatory, ParameterSetName = 'VaaS')]
+        [ValidateSet('System Admin', 'PKI Admin', 'Resource Owner', 'Guest')]
+        [string] $Role,
+
+        [Parameter(Mandatory, ParameterSetName = 'TPP')]
+        [ValidateSet('TLS', 'SSH', 'Code Signing')]
+        [string[]] $Product,
+
+        [Parameter(ParameterSetName = 'TPP')]
+        [string[]] $Description,
```
Comment (astro2000j) [inline VenafiPS/Public/New-VenafiTeam.ps1:104]:
```suggestion
        [string] $Description,
```

Comment (astro2000j) [reply VenafiPS/Public/New-VenafiTeam.ps1:104]: TPP WebSDK expects string for description

File: VenafiPS/Public/Get-VenafiIdentity.ps1 [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+.PARAMETER ID
+For TPP this is the guid or prefixed universal id.  To search, use Find-TppIdentity.
+For VaaS this can either be the user id (guid) or username which is the email address.
+
+.PARAMETER IncludeAssociated
+Include all associated identity groups and folders.  TPP only.
+
+.PARAMETER IncludeMembers
+Include all individual members if the ID is a group.  TPP only.
```
Comment (astro2000j) [inline VenafiPS/Public/Get-VenafiIdentity.ps1:18]: Are the `-IncludeAssociated` and `-IncludeMembers` specific to local TPP Teams? If so, this should be documented.

Comment (gdbarron) [reply VenafiPS/Public/Get-VenafiIdentity.ps1:18]: It's documented as 'TPP only' above.  I've also added a warning message if those params are used with vaas.

**Pull request merged**